File: RB101_Programming_Foundations/Small Probems/Easy7/Python/capitalize_words.py
# Write a method that takes a single String argument and returns a new string that contains the original value of the argument with the first character of every word capitalized and all other letters lowercase.

# You may assume that words are any sequence of non-blank characters.

'''
Input: a string

Output: original string titlecased

Rules


Data structure / Algorithms
    split the string into words
    for each word, capitalize it
    join the array by space


'''

def word_cap(string):
    words = string.split()

    for index, word in enumerate(words):
        words[index] = word.capitalize()
    return ' '.join(words)

print(word_cap('four score and seven') == 'Four Score And Seven')
print(word_cap('the javaScript language') == 'The Javascript Language')
print(word_cap('this is a "quoted" word') == 'This Is A "quoted" Word')

File: RB101_Programming_Foundations/small_problems/Easy7/capitalize_words.rb
# input: string
# output : new string with original string's individual word titlecased

require 'pry'

def word_cap(string)
  words_array = string.split(/[^A-Za-z0-9"']/).map do |word|
    word[0] = word[0].upcase
    word[1..-1] = word[1..-1].downcase
    word
  end

  words_array.join(' ')
end



p word_cap('four score and seven') == 'Four Score And Seven'
p word_cap('the javaScript language') == 'The Javascript Language'
p word_cap('this is a "quoted" word') == 'This Is A "quoted" Word'